File: backend/src/Core/Entities/Favorite.cs
namespace Core.Entities
{
    public class Favorite
    {
        public int Id { get; set; }
        public int MobileUserId { get; set; }
        public int ProductId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Relations
        public MobileUser MobileUser { get; set; }
        public Product Product { get; set; }
    }
} 

File: mobile/lib/features/products/domain/models/favorite.dart
import 'product.dart';

class Favorite {
  final int id;
  final int productId;
  final DateTime createdAt;
  final Product product;

  Favorite({
    required this.id,
    required this.productId,
    required this.createdAt,
    required this.product,
  });

  // Convertir JSON en objet Favorite
  factory Favorite.fromJson(Map<String, dynamic> json) {
    return Favorite(
      id: json['id'],
      productId: json['productId'],
      createdAt: DateTime.parse(json['createdAt']),
      product: Product.fromJson(json['product']),
    );
  }

  // Convertir objet Favorite en JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'productId': productId,
      'createdAt': createdAt.toIso8601String(),
      'product': product.toJson(),
    };
  }

  @override
  String toString() {
    return 'Favorite{id: $id, productId: $productId, createdAt: $createdAt}';
  }
} 